### ISSUE TITLE
[Bug] Only 1 AsusRouter device appears when having multiple instances of the integration

### ISSUE THREAD

**Issue by mihsu81** [Vaskivskyi/ha-asusrouter#527]
### The problem

Currently when adding multiple instances of the AsusRouter integration, only 1 device gets created which contains the entities of all the other devices. This makes it difficult to look for a device and its entities.
It also shows the AiMesh nodes appear as connected through another device than the router.
![image](https://user-images.githubusercontent.com/40715497/217367363-35b86cbb-90ea-4dbd-889c-8ac4c7992b78.png)

![image](https://user-images.githubusercontent.com/40715497/217366833-7d345faf-e515-47d3-8b1a-0811d394f9ca.png)

![image](https://user-images.githubusercontent.com/40715497/217367220-eeb05259-51f3-4c0a-bd1c-6c64c3f4e52f.png)

### Your device model

RT-AX88U

### Firmware type

Merlin

### Firmware version

3.0.0.4.388.1

### Integration version with the issue

0.19.0

### Method of the integration installation

HACS

### What version and type of Home Assistant installation do you use

2023.2.2

**Comment by Vaskivskyi** [comment]
That's extremely weird. Was the behaviour the same with previous versions or only with 0.19.0? Also, did you update to 0.19.0 with integration being set up or did you update and then configured the integration?

**Comment by mihsu81** [comment]
Always been like this. I did recreate the integrations about 3 months ago but kept coming back with the same layout.

**Comment by Vaskivskyi** [comment]
Nope. Actually, this is a bug. There was a similar one, but it was already fixed some time ago

If you don't mind reconfiguring the integration, you might try to remove it completely and configure it again to see whether it will configure properly. This might also help with the other issue #528 you have

If you cannot reconfigure it, we can try to find another solution

**Comment by mihsu81** [comment]
Looks like deleting all devices / integrations and adding them again doesn't help.
The entities of the previously added devices appear as entities of the last device i add.

In the below case the Router's entities appear now as the entities of the Kitchen AP device which I have added as an AiMesh node.

![image](https://user-images.githubusercontent.com/40715497/217561851-7c824c41-5baf-4943-aa47-42e6a2a170f9.png)

**Comment by RenierM26** [comment]
Hi There, I have two XT12 and two GT-AX11000. I get the same issue after adding the 2nd GT-AX11000.

Some observations:
1) It's not happening with the two XT12 AIMesh nodes.
2) The one GT-AX11000 is my main router the other is a mesh node. 

This leads me to think it's some or other issue with the device "device_info" property possibly a combination with the regular expression in def to_unique_id(raw: str). I'm also getting an entity does not have uniqueid error in HASS.

Hass dev docs on how entity/device mappings are made.
https://developers.home-assistant.io/docs/device_registry_index/

For formatting a MAC as unique is, HASS provides a built in helper "homeassistant.helpers.device_registry.format_mac"

https://developers.home-assistant.io/docs/entity_registry_index?_highlight=format&_highlight=mac#example-acceptable-sources-for-a-unique-id

**Comment by mihsu81** [comment]
I've changed some of the devices in my network and now I have x1 RT-AX88U Pro as a router, and as AiMesh nodes x2 TUF-AX5400 and the previous router RT-AX88U. The Media Bridge RT-AC87U keeps getting linked to the RT-AX88U appearing as the same device. For a while, I also had the RT-AC87U and the RT-AX88U as media bridges and again they would appear as 1 device.
I even factory reset all 5 devices using the WPS 30/30/30 method and completely removed the Asusrouter integration and rebooted HA. It didn't change anything.

**Comment by modelstudent101** [comment]
Have this same issue:
Router: AX-86U [Merlin 388.2]
Access Point: AX-82U [Gnuton-Merlin 388.2.2]

AX-86U device integration functions normally, but adding the AX-82U causes both to populate as a single device (device information reflects AX-82U).

- "Devices" page only shows AX-82U, and firmware.
- "Entities" page only shows one set of sensors and switches reporting.
- Router control is enabled on both routers.

**Comment by Vaskivskyi** [comment]
Unfortunately, I didn't manage to find any issue in the code itself. I will try to get access to an additional device to test the multi-device behaviour in real-life conditions. Apparently, my test sets for AiMesh with 2 devices are not universal enough

**Comment by Vaskivskyi** [comment]
With version `0.40.0` (already available in HACS), this issue should be fixed.

Devices should not get merged anymore. And if the automated fix works for your case, the old bugged device will be automatically removed and new one(s) created.

In case it does not work just after the restart of HA, please reload the integration once more manually before (re)opening an issue. In the worst-case scenario, you might need to manually remove devices, restart the HA and set up integration again.
